pkg/controller/psbackup/controller.go

```diff
@@ -333,6 +337,63 @@ func (r *PerconaServerMySQLBackupReconciler) getBackupSource(ctx context.Context
 	return source, nil
 }
 
+type Topology struct {
+	Primary  string
+	Replicas []string
+}
+
+func (r *PerconaServerMySQLBackupReconciler) getTopology(ctx context.Context, cluster *apiv1alpha1.PerconaServerMySQL, operatorPass string) (Topology, error) {
```

Comment:
I don't like the idea of removing the `pkg/mysql/topology` package and adding a new method to the reconciler. This could potentially make the reconciler excessively complex and difficult to refactor when making changes related to topology, such as the ones proposed in [this](https://github.com/percona/percona-server-mysql-operator/pull/339) PR.

I think the reconciler should have methods that are only responsible for reconciling the cluster, without retrieving any data from the database. The `pkg/db` package is more suitable for that purpose.

If the topology is only used for the `pkg/controller/psbackup` package, then let's move `pkg/mysql/topology` to the `pkg/controller/psbackup/topology`

Reply:
I see your point, but why I don't necessarily agree is that of having a whole package for only one thing used in one place. And the reconciler is calling the database all over the place, the way it manages the cluster involves calling DB, so I don't see the problem with this in that sense.

I figured out that you introduced this topology package because of the proposed PR, but what I suggest is that we don't design our code for the future that might never happen, it can just complicate things uneccesarily. When we need it we can introduce whatever we want.

Reply:
I agree that it's too much to create a separate package. But I don't want to add another method to reconciler, even if it's simple. We can try to make it a function instead of a method. We can also just put this function in a separate file, such as `pkg/mysql/psbackup/topology.go`. What do you think?

Actually, I don't think my suggestion is worth wasting time on. This is not a critical problem, so I'll approve the PR as soon as @egegunes' suggestions are resolved.

Reply:
I agree with you Andrii, getting the topology from the reconciler is not that nice, like I said, better maybe to have it in a function. But even if we have it it will not stick out, we have something similar with `isGRReady` on `PerconaServerMySQLReconciler`.

Reply:
[8831094](https://github.com/percona/percona-server-mysql-operator/pull/517/commits/8831094897ddce773e3db6ed3000b3a890470134)